ares.h

```diff
@@ -368,8 +368,8 @@ struct iovec;
 struct ares_socket_functions {
    ares_socket_t(*asocket)(int, int, int, void *);
    int(*aclose)(ares_socket_t, void *);
-   int(*aconnect)(ares_socket_t, const struct sockaddr *, socklen_t, void *);
-   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, socklen_t *, void *);
+   int(*aconnect)(ares_socket_t, const struct sockaddr *, ares_socklen_t, void *);
+   ssize_t(*arecvfrom)(ares_socket_t, void *, size_t, int, struct sockaddr *, ares_socklen_t *, void *);
    ssize_t(*asendv)(ares_socket_t, const struct iovec *, int, void *);
```

Comment:
IMO there is another problem with this lines of code, `ssize_t` is not C standard (POSIX?) and is defined for Windows in config-win32.h which gets not included into `ares.h`